### PR TITLE
Buttons Lua object

### PR DESCRIPTION
**Pull request by CerBor**
In this PR I suggest to add new Buttons object for controlling ```fight, act, item, mercy``` buttons.

I wanted to make battle, when enemy can move your buttons, but I didn't find any functions for it, so I decided to write my own.
So, with this PR you can move, \*block\* and fully control buttons.

I hope that it will be useful for many players.

### CODE REVIEW

File: Assets/Scripts/Battle/UIController.cs [hunk 1]
```diff
@@ -1041,6 +1043,48 @@ public enum UIState {
             PlaySound(AudioClipRegistry.GetSound("menuconfirm"));
     }
 
+    public static void BlockButton(string btn)
+    {
```
Comment (RhenaudTheLukark) [inline Assets/Scripts/Battle/UIController.cs:1047]: Please follow the same coding style as the rest of the engine (curly braces on the same line as the block using it, etc)

File: Assets/Scripts/Lua/CLRBindings/LuaButtons.cs [hunk 2]
```diff
@@ -0,0 +1,154 @@
+﻿using UnityEngine;
+
+public class LuaButtons
```
Comment (RhenaudTheLukark) [inline Assets/Scripts/Lua/CLRBindings/LuaButtons.cs:3]: Is it possible for me to ask you to add the buttons to the newly created UI object instead? That way it's all in the same object, and buttons *are* part of the UI, after all.

**Review by RhenaudTheLukark — CHANGES_REQUESTED** [review]
It's interesting! I'd love to get this PR, however I very recently added an object called the UI object and I think it would be better to move your addition in it!

Other than that and a few coding style mismatches, I think it's a good PR!

File: Assets/Scripts/Battle/UIController.cs (hunk 1 above)
Comment (CerBor) [reply Assets/Scripts/Battle/UIController.cs:1047]: Hello. I didn't know about the braces, will fix it soon

File: Assets/Scripts/Lua/CLRBindings/LuaButtons.cs (hunk 2 above)
Comment (CerBor) [reply Assets/Scripts/Lua/CLRBindings/LuaButtons.cs:3]: 👌 ok

**Comment by AllyTally** [comment]
While I don't agree with the design decisions in this PR, it's fine for now.
However I'd recommend calling the block/unblock functions disable/enable instead.

**Comment by CerBor** [comment]
OK, will do it too

**Comment by RhenaudTheLukark** [comment]
Alright, the current workload I have has been cleared, so I can accept this commit.

I don't suppose you've added all this to the docs, but I'll do it myself.

I also plan to clean up some things, but it shouldn't take me too long so it's alright.

Thanks a ton for the contribution!

**Pull request merged**